C/src/TaskSetGenerator/gui.py

```diff
@@ -0,0 +1,223 @@
+import tkinter as tk
+from tkinter import ttk
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+from scipy.interpolate import interp1d
+import numpy as np
+
+import matplotlib
+matplotlib.use('tkagg')
+
+import os
+import sys
+import subprocess
+import shutil
+from functools import partial
+import random
+
+timeUnits = ['sec', 'msec', 'usec', 'nsec']
+
+def step(num_workers, scheduler_type):
+
+    global char_to_replace, template
+    char_to_replace['SCHEDULER_TYPE'] = scheduler_type
+    char_to_replace['NUM_WORKERS'] = str(num_workers)
+
+    contents = template
+    for key, value in char_to_replace.items():
+        contents = contents.replace(key, value)
+
+    filename = f'practice_{scheduler_type}_{num_workers}'
+    filepath = f'{WORKING_DIR}/.gui/src/{filename}.lf'
+    with open(filepath, "w") as lf_file:
+        lf_file.write(contents)
+        lf_file.close()
+        print(f"File saved: {filepath}")
+
+    out = subprocess.run([f'./gradlew','runLfc', '--args', filepath], capture_output=True)
+
+    print("Build successfully!")
+    lf_out = subprocess.run([f'{WORKING_DIR}/.gui/bin/{filename}'], capture_output=True)
+
+    for line in reversed(lf_out.stdout.decode("utf-8").split('\n')):
+        if line.startswith("---- Elapsed physical"):
+            exe_time = line.split(' ')[-1]
+            break
+
+    exe_time = int(exe_time.replace(',','')) / 1000000000
+    return exe_time
+    
+
+def runLfc():
+
+    global char_to_replace, template
+    workers = [1, 2, 5, 20]
+    schedulers = ["NP", "GEDF_NP", "GEDF_NP_CI"]
+    
+
+    char_to_replace = {
+        'SCHEDULER_TYPE': 'NP',
+        'NUM_TASKS': str(numOfTasks.get()),
+        'TOTAL_TIME': str(totalTime.get()) + " " + str(totalTimeUnit.get()),
+        'UTILIZATION': utilization.get(),
+        'NUM_WORKERS': '1',
+        'PERIODIC':'false' if periodicity.get() == 1 else 'true',
+        'PERIOD': str(period.get()) + " " + str(periodUnit.get())
+    }
+
+    with open(TEMPLATE_PATH) as f:
+        template = f.read()
+
+    exe_times = {}
+
+    workers = [i for i in range(1, 21)]
+    for scheduler in schedulers:
+        exe_time = []
+        for worker in workers:
+            exe_time.append(step(worker, scheduler))
+        exe_times[scheduler] = exe_time.copy()
+        exe_time.clear()
+
+    result = {}
+    result['workers'] = workers
+    result['exe_times'] = exe_times
+    
+    plot_graph(result)
+    #print(result)
+
+
+def plot_graph(data):
+    
+    fig, ax = plt.subplots()
+    plt.axis([1, 25, 0.0, 5.0])
+
+    workers = data['workers']
+    scatters = [1, 2, 5, 20]
+
+    axes = []
+    patches = []
+    schedulers = data['exe_times'].keys()
+    for scheduler in schedulers:
+        
+        color = (random.random(), random.random(), random.random())
```

Comment:
I think this approach is not ideal. The colors can often be very close to each other and they change every time. I think using a preset of colors such as [this](https://davidmathlogic.com/colorblind/) is better.

Reply:
I fully agree with the problem of the approach. Actually, I have seen several cases where the colors are very similar so it is hard to distinguish graphs. I will fix the approach using a preset of colors like your suggestion